l10n_br_nfe/models/document_line.py

```diff
@@ -173,8 +173,8 @@ def _export_fields_nfe_40_prod(self, xsd_fields, class_obj, export_dict):
         nfe40_cProd = self.product_id.default_code or self.nfe40_cProd or ""
         export_dict["cProd"] = nfe40_cProd
 
-        nfe40_xProd = self.product_id.name or self.name or ""
-        export_dict["xProd"] = nfe40_xProd[:120].replace("\n", "").strip()
+        nfe40_xProd = self.name or self.product_id.display_name or ""
```

Comment:
Não deveria usar o display_name, porque quando um produto tem o campo referencia interna aparece no display_name

```python
>>> product.display_name
'[FURN_0269] Office Chair Black'
>>> product.name
'Office Chair Black'
```

Reply:
Você deveria usar a chave display_default_code para pegar a descrição no caso das variantes mas sem a referência interna

```python
>>> product.with_context(display_default_code=False).display_name
'Office Chair Black'
```

Reply:
Agora sobre a troca da ordem de pegar primeiro o self.name e depois pegar o self.product_id.display_name, não acho correta, porque caso seja necessário adicionar alguma informação sobre a descrição do produto, pode ser adicionado na observação da linha da nota fiscal (essa observação sai abaixo da descrição do produto). Também trocando para usar o displany_name, pode ser usada a descrição comercial no cadastro de produtos.

Reply:
Boa @renatonlima! não tinha me atentado ao `default_code` que vinha junto, legal que já tem uma funcionalidade nativa para suprimir isso. 

Quanto ao por o `self.name` antes do produto eu tinha pensado na passabilidade de algum usuário desejar fornecer uma descrição alternativa para a fatura em questão ao que está no cadastro do produto, mas confesso que de fato isso não é um problema para nós, então removi essa alteração desta PR.

Nosso objeto principal é a correção da descrição quando o produto for uma variante.
Então só com a troca ali pelo `display_name` já vamos ficar bastante satisfeitos.
Fiz a correção para incluir o `display_default_code=False` conforme sua sugestão.

Obrigado pela revisão!